scripts/python/ip_route_get_to.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+# Copyright 2018 IBM Corp.
+#
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import nested_scopes, generators, division, absolute_import, \
+    with_statement, print_function, unicode_literals
+
+import argparse
+import netaddr
+import socket
+from pyroute2 import IPRoute
+
+from lib.exception import UserException
+import lib.logger as logger
+
+
+def ip_route_get_to(host):
+    """Get interface IP that routes to hostname or IP address
+
+    Args:
+        host (str): Hostname or IP address
+
+    Returns:
+        str: Interface IP with route to host
+
+    Raises:
+        UserException: Hostname will not resolve or no route to host
+    """
+    log = logger.getlogger()
+
+    # Check if host is given as IP address
+    if netaddr.valid_ipv4(host, flags=0):
+        host_ip = host
+    else:
+        try:
+            host_ip = socket.gethostbyname(host)
+        except socket.gaierror as exc:
+            log.debug("Unable to resolve host to IP: '{}' exception: '{}'"
+                      .format(host, exc))
+            raise UserException("Unable to resolve hostname '{}'!"
```

Comment:
How will Ansible deal with this exception?

Reply:
I will print out a wall of red text :). I think the only way for Ansible to do more is to ignore errors, register results and add additional tasks. Presumably the 'software_hosts' inventory has been validated _before_ running this play so maybe it's safe to just remove this 'raise'.

Reply:
My concern is that we shouldn't stop execution if one of the hosts is not accessible. So I agree it would be better to remove the raise, and maybe change the log level to warning.